sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp

```diff
@@ -259,20 +261,230 @@ AttestationAdministrationClient::ResetAttestationPolicy(
+}
+
+Azure::Response<Models::AttestationToken<PolicyCertificateListResult>>
+AttestationAdministrationClient::GetPolicyManagementCertificates(
+    GetPolicyManagementCertificatesOptions const& options,
+    Azure::Core::Context const& context) const
+{
+  auto request = AttestationCommonRequest::CreateRequest(
+      m_endpoint, m_apiVersion, HttpMethod::Get, {"certificates"}, nullptr);
+
+  // Send the request to the service.
+  auto response = AttestationCommonRequest::SendRequest(*m_pipeline, request, context);
+
+  // Deserialize the Service response token and return the JSON web token returned by the
+  // service.
+  std::string responseToken = AttestationServiceTokenResponseSerializer::Deserialize(response);
+
+  // Parse the JWT returned by the attestation service.
+  auto resultToken = AttestationTokenInternal<
+      Models::_detail::GetPolicyCertificatesResult,
+      PolicyCertificateGetResultSerializer>(responseToken);
+
+  // Validate the token returned by the service. Use the cached attestation signers in the
+  // validation.
+  std::vector<AttestationSigner> const& signers = GetAttestationSigners(context);
+  resultToken.ValidateToken(
+      options.TokenValidationOptions ? *options.TokenValidationOptions
+                                     : this->m_tokenValidationOptions,
+      signers);
+
+  Models::_detail::JsonWebKeySet jwks(
+      *static_cast<AttestationToken<Models::_detail::GetPolicyCertificatesResult>>(resultToken)
+           .Body.PolicyCertificates);
+  Models::PolicyCertificateListResult returnedResult;
+  for (const auto& certificate : jwks.Keys)
   {
-    returnedResult.PolicySigner = AttestationSignerInternal(*internalResult.PolicySigner);
+    returnedResult.Certificates.push_back(AttestationSignerInternal(certificate));
   }
-  if (internalResult.PolicyTokenHash)
+
+  // Construct a token whose body is the get policy certificates result, but whose token is the
+  // response from the service.
+  auto returnedToken = AttestationTokenInternal<Models::PolicyCertificateListResult>(
+      responseToken, returnedResult);
+  return Response<AttestationToken<Models::PolicyCertificateListResult>>(
+      returnedToken, std::move(response));
+}
+
+Azure::Response<Models::AttestationToken<Models::PolicyCertificateModificationResult>>
+AttestationAdministrationClient::AddPolicyManagementCertificate(
+    std::string const& pemEncodedX509CertificateToAdd,
+    AttestationSigningKey const& existingSigningKey,
+    AddPolicyManagementCertificatesOptions const& options,
+    Azure::Core::Context const& context) const
+{
+  // Calculate a signed (or unsigned) attestation policy token to send to the service.
+  // Embed the encoded policy in the StoredAttestationPolicy.
+  auto x5cToAdd(Cryptography::ImportX509Certificate(pemEncodedX509CertificateToAdd));
+
+  // Create a JWK to add to the body.
+  JsonWebKey jwkToSend;
+  jwkToSend.kty = x5cToAdd->GetKeyType();
+  jwkToSend.x5c = std::vector<std::string>();
```

Comment:
Are these public fields from the JsonWebKey?  If yes, it should be UpperCase like `jwkToSend.X5c = ....`  .  And the same for Kty